ouroboros/shared/value.go

```diff
@@ -53,20 +53,64 @@ func Subtract(a Value, b Value) Value {
 }
 
 func Enough(have Value, want Value) (bool, error) {
+	var ErrInsufficientFunds = fmt.Errorf("insufficient funds")
```

Comment:
best practice is to make this public, so people can check for it by using `errors.Is(...)`; as a local variable to this function, people wouldn't have any variable to reference.